plugins/@grouparoo/app-templates/src/source/query/profileProperty.ts

```diff
@@ -0,0 +1,77 @@
+import {
+  ProfilePropertyPluginMethod,
+  plugin,
+  ProfilePropertyPluginMethodResponse,
+} from "@grouparoo/core";
+import {
+  ExecuteQueryMethod,
+  ValidateQueryMethod,
+  queryKey,
+} from "./pluginMethods";
+
+export interface GetProfilePropertyMethod {
+  (argument: {
+    executeQuery: ExecuteQueryMethod;
+    validateQuery?: ValidateQueryMethod;
+  }): ProfilePropertyPluginMethod;
+}
+
+export const getProfileProperty: GetProfilePropertyMethod = ({
+  executeQuery,
+  validateQuery,
+}) => {
+  const profileProperty: ProfilePropertyPluginMethod = async ({
+    connection,
+    appOptions,
+    profile,
+    profilePropertyRule,
+    profilePropertyRuleOptions,
+  }) => {
+    const ruleQuery = profilePropertyRuleOptions[queryKey];
+    const query = await plugin.replaceTemplateProfileVariables(
+      ruleQuery,
+      profile
+    );
+    if (!validateQuery) {
+      validateQuery = validateGenericQuery;
+    }
+    validateQuery({ query });
+
+    let response: ProfilePropertyPluginMethodResponse;
+    try {
+      // Run the query
+      const rows = await executeQuery({ connection, appOptions, query });
+
+      // Get the results
+      if (rows && rows.length > 0) {
+        if (!profilePropertyRule.isArray) {
+          const row = rows[0];
+          response = [Object.values(row)[0]];
+        } else {
+          response = rows.map((row) => Object.values(row)[0]);
+        }
+      }
+    } catch (error) {
+      throw new Error(`Error with Query: - \`${query}\`, Error - ${error}`);
```

Comment:
Since this is going to be used generically, the error message might want to include the source + app names/guids